automation/src/test/scala/org/broadinstitute/dsde/test/api/orch/AsyncTestUtils.scala

```diff
@@ -0,0 +1,79 @@
+package org.broadinstitute.dsde.test.api.orch
+
+import org.scalatest.Assertion
+import org.scalatest.concurrent.Eventually
+import org.scalatest.concurrent.PatienceConfiguration.{Interval, Timeout}
+import org.scalatest.exceptions.TestFailedException
+
+import javax.xml.parsers.FactoryConfigurationError
+
+/**
+  * Some helper functions to get asynchronous tests failing quickly
+  * Should some day be made available to a wider codebase
+  */
+trait AsyncTestUtils extends Eventually {
+
+  /**
+    * Wraps a function in an eventually wrapped in a try-catch;
+    * Functions as an eventually but will fail test without retrying if test-ending errors thrown in the test function
+    * @param timeout
+    * @param interval
+    * @param testFunction Test to run; should include a failAndEscape (could be within assertWithEscape)
+    * @tparam T the result of the testFunction
+    * @return The result of the testFunction (an assertion) if an error is not thrown
+    */
+  def eventuallyWithEscape[T](timeout: Timeout, interval: Interval, testFunction: => T): T =
+    try {
+      eventually(timeout, interval){
+        testFunction
+      }
+    } catch {
+      case ex:FactoryConfigurationError => throw new TestEscapedException(ex.getMessage)
+    }
+
+  /**
+    * eventuallyWithEscape but the eventually uses implicit timeout and interval
+    * @param testFunction Test to run; should include a failAndEscape (could be within assertWithEscape)
+    * @tparam T the result of the testFunction
+    * @return The result of the testFunction (an assertion) if an error is not thrown
+    */
+  def eventuallyWithEscape[T](testFunction: => T): T =
+    try {
+      eventually{
+        testFunction
+      }
+    } catch {
+      case ex:FactoryConfigurationError => throw new TestEscapedException(ex.getMessage)
+    }
+
+  /**
+    * Will fail a test that uses eventually instead of retrying
+    * @param message Error message
+    * @return
+    */
+  def failAndEscape(message: String): Nothing = { throw new FactoryConfigurationError(message) }
```

Comment:
not necessary for this PR, because this is all rather prospective at this point: I could see some point in the future also adding a 
```scala
def failAndEscape(exception: Exception): Nothing = throw new FactoryConfigurationError(exception)
```
method that allows the original exception to propagate. Of course this would need some corresponding support in `eventuallyWithEscape` to check if the `FactoryConfigurationError` contains an exception and rethrow it.